tests/extras/datasets/spark/test_deltatable_dataset.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from delta import DeltaTable
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, StringType, StructField, StructType
+
+from kedro.extras.datasets.spark import DeltaTableDataset, SparkDataSet
+
+
+@pytest.fixture(autouse=True)
+def delta_spark_session(spark_session):
+    return spark_session
+
+
+@pytest.fixture
+def sample_spark_df():
+    schema = StructType(
+        [
+            StructField("name", StringType(), True),
+            StructField("age", IntegerType(), True),
+        ]
+    )
+
+    data = [("Alex", 31), ("Bob", 12), ("Clarke", 65), ("Dave", 29)]
+
+    return SparkSession.builder.getOrCreate().createDataFrame(data, schema)
+
+
+class TestDeltaTableDataSet:
+    def test_load(self, tmp_path, sample_spark_df):
+        filepath = (tmp_path / "data.csv").as_posix()
```

Comment:
Good start - not sure however why this `csv` one is here? Or is it to be removed?

Reply:
Actually if you need data you could just `convertToDelta` in place?